### PR TITLE
Quick solution for casting result primary keys to the correct type

### LINKED ISSUE #5881: 3.0 - Incorrect data type for primary key after persisting new entity
After a new entity is persisted for an auto increment primary key field, the value set for corresponding entity property is of type string instead of integer.

@lorenzo mentioned on IRC that he's aware of the problem and already dreamt about a possible solution the night earlier (no kidding, he's that damm good).

### PR DESCRIPTION
**Pull request by lorenzo**
Fixes #5881

### CODE REVIEW

File: src/ORM/Table.php [hunk 1]
```diff
@@ -1505,10 +1505,13 @@ protected function _insert($entity, $data)
...
             $success = $entity;
             $entity->set($filteredKeys, ['guard' => false]);
+            $schema = $this->schema();
+            $driver = $this->connection()->driver();
             foreach ($primary as $key => $v) {
                 if (!isset($data[$key])) {
                     $id = $statement->lastInsertId($this->table(), $key);
-                    $entity->set($key, $id);
+                    $type = $schema->columnType($key);
+                    $entity->set($key, Type::build($type)->toPHP($id, $driver));
```
Comment (markstory) [inline src/ORM/Table.php]: Aren't the most common cases going to be int/string? We could us is_numeric()/ctype_digit() to handle int casing.

Comment (lorenzo) [inline src/ORM/Table.php]: Those are definitely the most common cases, but why treat this case differently than the rest?

Comment (markstory) [inline src/ORM/Table.php]: Fair. I guess the overhead of getting the type object won't be that high.

Comment (lorenzo) [inline src/ORM/Table.php]: Not really, the type will be constructed at that point already

**Pull request merged**